tests/bgp/test_bgp_bbr.py

```diff
@@ -0,0 +1,253 @@
+'''This script is to test the BGP Bounce Back Routing (BBR) feature of SONiC.
+'''
+import json
+import logging
+import time
+
+from collections import defaultdict
+from collections import namedtuple
+
+import pytest
+import requests
+import ipaddr as ipaddress
+
+from jinja2 import Template
+from natsort import natsorted
+from tests.common.helpers.assertions import pytest_assert
+from tests.common.helpers.parallel import reset_ansible_local_tmp
+from tests.common.helpers.parallel import parallel_run
+
+pytestmark = [
+    pytest.mark.topology('t1'),
+    pytest.mark.device_type('vs')
+]
+
+logger = logging.getLogger(__name__)
+
+
+EXABGP_BASE_PORT = 5000
+EXABGP_BASE_PORT_V6 = 6000
+
+
+BBR_PREFIX = '172.16.10.0/24'
+BBR_PREFIX_V6 = '2000:0:172:16:10::/96'
+
+DUMMY_ASN1 = 64101
+DUMMY_ASN2 = 64102
+
+
+@pytest.fixture
+def disable_enable_bbr(duthosts, rand_one_dut_hostname):
+    duthost = duthosts[rand_one_dut_hostname]
+    bgp_bbr_config = Template(open("./bgp/templates/bgp_bbr_config.json.j2").read())
+
+    duthost.copy(content=bgp_bbr_config.render(BGP_BBR_STATUS='disabled'), dest='/tmp/disable_bbr.json')
+    duthost.copy(content=bgp_bbr_config.render(BGP_BBR_STATUS='enabled'), dest='/tmp/enable_bbr.json')
+
+    logger.info('Disable BGP_BBR')
+    duthost.shell('sonic-cfggen -j /tmp/disable_bbr.json -w ')
+    time.sleep(3)
+    yield
+
+    logger.info('Enable BGP_BBR')
+    duthost.shell('sonic-cfggen -j /tmp/enable_bbr.json -w')
+
+
+@pytest.fixture(scope='module')
+def setup(duthosts, rand_one_dut_hostname, tbinfo, nbrhosts):
+    duthost = duthosts[rand_one_dut_hostname]
+    if tbinfo['topo']['type'] != 't1':
+        pytest.skip('Unsupported topology type: {}, supported: {}'.format(tbinfo['topo']['type'], 't1'))
+
+    mg_facts = duthost.get_extended_minigraph_facts(tbinfo)
+
+    tor_neighbors = natsorted([neighbor for neighbor in nbrhosts.keys() if neighbor.endswith('T0')])
+    t2_neighbors = [neighbor for neighbor in nbrhosts.keys() if neighbor.endswith('T2')]
+    tor1 = tor_neighbors[0]
+    other_vms = tor_neighbors[1:] + t2_neighbors
+
+    neigh_peer_map = defaultdict()
+    for bgp_neigh in mg_facts['minigraph_bgp']:
+        name = bgp_neigh['name']
+        peer_addr = bgp_neigh['peer_addr']
+        if name not in neigh_peer_map:
+            neigh_peer_map[name] = defaultdict()
+        if ipaddress.IPAddress(peer_addr).version == 4:
+            neigh_peer_map[name].update({'peer_addr': peer_addr})
+        else:
+            neigh_peer_map[name].update({'peer_addr_v6': peer_addr})
```

Comment:
In case of defaultdict your code would something like
```
neigh_peer_map = defaultdict(dict)
for bgp_neigh in mg_facts['minigraph_bgp']:
        name = bgp_neigh['name']
        peer_addr = bgp_neigh['peer_addr']
        if ipaddress.IPAddress(peer_addr).version == 4:
            neigh_peer_map[name].update({'peer_addr': peer_addr})
        else:
            neigh_peer_map[name].update({'peer_addr_v6': peer_addr})
```

Reply:
Thanks for the comments! Updated.